packages/events/src/router/middleware/authorization/index.ts

```diff
@@ -38,3 +41,29 @@ export function requiresAnyOfScopes(scopes: Scope[]) {
     throw new TRPCError({ code: 'FORBIDDEN' })
   }
 }
+
+/**
+ * Middleware to check if a record can be unassigned.
+ * A record can only be unassigned in one of the following cases:
+ *  - The record is assigned to the user attempting to unassign it.
+ *  - The user attempting to unassign the record has the scope: ${SCOPES.RECORD_UNASSIGN_OTHERS}.
+ * @returns A TRPC-compatible middleware function.
+ */
+export function canUnassign() {
```

Comment:
Can we just make this a regular function and not a middleware? I think it will only be used in `packages/events/src/router/event/index.ts`. Middlewares are handy for preprocessing requests but from they also add a lot of indirection